processors/ratelimiter/ratelimiterprocessor.go

```diff
@@ -0,0 +1,126 @@
+package ratelimiter
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	pb_struct "github.com/envoyproxy/go-control-plane/envoy/extensions/common/ratelimit/v3"
+	pb "github.com/envoyproxy/go-control-plane/envoy/service/ratelimit/v3"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/metadata"
+)
+
+var _ component.TracesProcessor = (*processor)(nil)
+
+var (
+	tagTenantID               = tag.MustNewKey("tenant-id")
+	droppedSpanCountSoftLimit = stats.Int64("tenant_id_dropped_span_count_soft_limit", "Number of spans dropped because of rate limiting per tenant due to soft limit", stats.UnitDimensionless)
+	droppedSpanCount          = stats.Int64("tenant_id_dropped_span_count", "Number of spans dropped per tenant due to global rate limiting", stats.UnitDimensionless)
+)
+
+func (p *processor) Start(_ context.Context, _ component.Host) error {
+	return nil
+}
+
+func (p *processor) Shutdown(_ context.Context) error {
+	return nil
```

Comment:
I think you can also put in the connection close and context cancelling here if you don't want to use options. Which means you have to track those objects in your processor.

Reply:
Done